src/fw-logs/fw-logs-xml-helper.cpp

```diff
@@ -1,309 +1,345 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
-#include "fw-logs-xml-helper.h"
-#include <string.h>
-#include <fstream>
-#include <iostream>
-#include <memory>
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
 
-using namespace std;
+#include <src/fw-logs/fw-logs-xml-helper.h>
+#include <src/librealsense-exception.h>
 
-namespace librealsense
+#include <rsutils/string/from.h>
+
+
+#include "../../third-party/rapidxml/rapidxml_utils.hpp"
+using namespace rapidxml;
+
+namespace librealsense {
+namespace fw_logs {
+
+std::vector< char > string_to_char_buffer( const std::string & xml_content )
 {
-    namespace fw_logs
-    {
-        fw_logs_xml_helper::fw_logs_xml_helper(string xml_content)
-            : _init_done(false),
-            _xml_content(xml_content)
-        {}
+    std::vector< char > buffer;
 
+    buffer.resize( xml_content.size() + 2 );
+    memcpy( buffer.data(), xml_content.data(), xml_content.size() );
+    buffer[xml_content.size()] = '\0';
+    buffer[xml_content.size() + 1] = '\0';
```

Comment:
Why do we need 2 of '\n' ?

Reply:
I don't know, it was like this in the previous implementation, I did not change.
`rapidxml` parses a zero-terminated xml string, but I did not dive into the docs and assumed that there may have been corner cases that required 2 `\0`.